test/middleware/express.test.ts

```diff
@@ -0,0 +1,341 @@
+            test('... Token Bucket', () => {
+                // FIXME: Is it possible to check which algorithm was chosen beyond error checking?
+                expect(
+                    expressRateLimitMiddleware(
+                        'TOKEN_BUCKET',
+                        { refillRate: 1, bucketSize: 10 },
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Leaky Bucket', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'LEAKY_BUCKET',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Fixed Window', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'FIXED_WINDOW',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Sliding Window', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'SLIDING_WINDOW_LOG',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Sliding Window Counter', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'SLIDING_WINDOW_COUNTER',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+        });
+
+        test('Throw an error for invalid schemas', () => {
+            const invalidSchema: GraphQLSchema = buildSchema(`{Query {name}`);
+
+            expect(
+                expressRateLimitMiddleware('TOKEN_BUCKET', {}, invalidSchema, { url: '' })
+            ).toThrowError('ValidationError');
+        });
+    });
+
+    describe('Middleware is Functional', () => {
+        // Before each test configure a new middleware amd mock req, res objects.
+        beforeAll(() => {
+            jest.useFakeTimers('modern');
+        });
+
+        afterAll(() => {
+            jest.useRealTimers();
+        });
+
+        beforeEach(() => {
+            middleware = expressRateLimitMiddleware(
+                'TOKEN_BUCKET',
+                { refillRate: 1, bucketSize: 10 },
+                schema,
+                {}
```

Comment:
Just checking, the `{}` on line `173` means no redis client provided and middleware will take care of it?

Reply:
What I had in mind was that it would default to connecting to a redis client at localhost and the default redis port.